### ISSUE TITLE
DataTable: checkIcon prop only applied for rows

### ISSUE THREAD

**Issue by BAISTM** [primefaces/primereact#5093]
### Describe the bug

The DataTable `checkIcon` is not applied on the header cell, it's only applied on the rows cell. Please look at the below screenshot

![image](https://github.com/primefaces/primereact/assets/129836651/b94b8a88-a15c-4b01-848b-632c8ae7823a)

### PrimeReact version

9.6.2

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

Chrome

### Steps to reproduce the behavior

Set the Datable `checkIcon` prop.

### Expected behavior

The `checkIcon` should also be applied on the header cell for more consistency.

**Comment by melloware** [comment]
Yep it looks like either the column or table needs a `selectAllCheckboxIcon`
